File: bin/system.lua
local function call_with_file(f, path, mode)
  local ____id = {io.open(path, mode)}
  local __h = ____id[1]
  local __e = ____id[2]
  if not __h then
    error(__e)
  end
  local __x1 = f(__h)
  __h.close(__h)
  return __x1
end
local function read_file(path)
  return call_with_file(function (f)
    return f.read(f, "*a")
  end, path)
end
local function write_file(path, data)
  return call_with_file(function (f)
    return f.write(f, data)
  end, path, "w")
end
local function file_exists63(path)
  local __f = io.open(path)
  local __id2 = is63(__f)
  local __e1 = nil
  if __id2 then
    local __r6 = is63(__f.read(__f, 0)) or 0 == __f.seek(__f, "end")
    __f.close(__f)
    __e1 = __r6
  else
    __e1 = __id2
  end
  return __e1
end
local function directory_exists63(path)
  local __f1 = io.open(path)
  local __id3 = is63(__f1)
  local __e2 = nil
  if __id3 then
    local __r8 = not __f1.read(__f1, 0) and not( 0 == __f1.seek(__f1, "end"))
    __f1.close(__f1)
    __e2 = __r8
  else
    __e2 = __id3
  end
  return __e2
end
local path_separator = char(_G.package.config, 0)
local function path_join(...)
  local __parts = unstash({...})
  return reduce(function (x, y)
    return x .. path_separator .. y
  end, __parts) or ""
end
local function get_environment_variable(name)
  return os.getenv(name)
end
local function stdout()
  return io.stdout
end
local function stderr()
  return io.stderr
end
local function write(x, out)
  local __out = out or stdout()
  __out.write(__out, x)
  return nil
end
local function exit(code)
  return os.exit(code)
end
local argv = nil
function set_argv(l)
  argv = l
  return argv
end
function get_argv()
  if nil63(argv) then
    set_argv(_G.arg or _G.args or {})
  end
  return argv
end
function opt63(x)
  return string63(x) and char(x, 0) == "-"
end
function parse_positional(args, pos)
  return cut(args, either(pos, 0), first(opt63, args, pos))
end
function parse_option(args)
  if opt63(hd(args)) then
    return {hd(args), parse_positional(args, 1)}
  end
end
function parse_arguments(aliases, argv)
  local __l = argv or get_argv()
  local __a = aliases or {}
  local __r21 = parse_positional(__l)
  __l = cut(__l, _35(__r21))
  while true do
    local __p = parse_option(__l)
    if not __p then
      break
    end
    local ____y = __p
    if yes(____y) then
      local ____id1 = ____y
      local __o = ____id1[1]
      local __args = ____id1[2]
      if __o == "--" then
        __l = cut(__l, 1)
        break
      end
      __l = cut(__l, 1 + _35(__args))
      local __e3 = nil
      if clip(__o, 0, 2) == "--" then
        __e3 = clip(__o, 2)
      else
        __e3 = clip(__o, 1)
      end
      local __k = __e3
      local __k1 = __a[__k] or __k
      local __e4 = nil
      if none63(__args) then
        __e4 = true
      else
        __e4 = __args
      end
      local __v = __e4
      __r21[__k1] = __v
      add(__r21, {__k1, __v})
    end
  end
  __r21.rest = __l
  set_argv(__r21.rest)
  return __r21
end
function arguments(aliases, argv)
  local __argv = argv or get_argv()
  local __r23 = parse_arguments(__argv, aliases)
  set_argv(__r23.rest)
  __r23.rest = nil
  if not empty63(__r23) then
    return __r23
  end
end
local function reload(module)
  package.loaded[module] = nil
  return require(module)
end
local function run(command)
  local __f2 = io.popen(command)
  local __x5 = __f2.read(__f2, "*all")
  __f2.close(__f2)
  return __x5
end
return {["read-file"] = read_file, ["write-file"] = write_file, ["file-exists?"] = file_exists63, ["directory-exists?"] = directory_exists63, ["path-separator"] = path_separator, ["path-join"] = path_join, ["get-environment-variable"] = get_environment_variable, stdout = stdout, stderr = stderr, write = write, exit = exit, ["get-argv"] = get_argv, ["set-argv"] = set_argv, arguments = arguments, reload = reload, run = run}

File: bin/system.js
var fs = require("fs");
var child_process = require("child_process");
var read_file = function(path) {
  return fs.readFileSync(path, "utf8");
};
var write_file = function(path, data) {
  return fs.writeFileSync(path, data, "utf8");
};
var file_exists63 = function(path) {
  return fs.existsSync(path, "utf8") && fs.statSync(path).isFile();
};
var directory_exists63 = function(path) {
  return fs.existsSync(path, "utf8") && fs.statSync(path).isDirectory();
};
var path_separator = require("path").sep;
var path_join = function(a, ..._42args) {
  return reduce(function(x, y) {
  return x + path_separator + y;
}, [..._42args], a);
};
var get_environment_variable = function(name) {
  return process.env[name];
};
var stdout = function() {
  return (process || io).stdout;
};
var stderr = function() {
  return (process || io).stderr;
};
var write = function(x, out) {
  return (out || stdout()).write(x);
};
var exit = function(code) {
  return process.exit(code);
};
var argv = undefined;
set_argv = function(l) {
  argv = l;
  return argv;
};
get_argv = function() {
  if (nil63(argv)) {
    set_argv(cut(process.argv, 2));
  }
  return argv;
};
var opt63 = function(x) {
  return string63(x) && char(x, 0) === "-" && !(x === "-");
};
parse_positional = function(args, pos) {
  return cut(args, either(pos, 0), first(opt63, args, pos));
};
parse_option = function(args) {
  if (opt63(hd(args))) {
    return [hd(args), parse_positional(args, 1)];
  }
};
parse_arguments = function(aliases, argv) {
  var __l = argv || get_argv();
  var __a = aliases || {};
  var __r18 = parse_positional(__l);
  __l = cut(__l, _35(__r18));
  while (true) {
    var __p = parse_option(__l);
    if (!__p) {
      break;
    }
    var ____y = __p;
    if (yes(____y)) {
      var ____id1 = ____y;
      var __op = ____id1[0];
      var __args = ____id1[1];
      if (__op === "--") {
        __l = cut(__l, 1);
        break;
      }
      __l = cut(__l, 1 + _35(__args));
      var __e1 = undefined;
      if (clip(__op, 0, 2) === "--") {
        __e1 = clip(__op, 2);
      } else {
        __e1 = clip(__op, 1);
      }
      var __k = __e1;
      var __k1 = __a[__k] || __k;
      var __e2 = undefined;
      if (none63(__args)) {
        __e2 = true;
      } else {
        __e2 = __args;
      }
      var __v = __e2;
      __r18[__k1] = __v;
      add(__r18, [__k1, __v]);
    }
  }
  __r18.rest = __l;
  set_argv(__r18.rest);
  return __r18;
};
arguments = function(aliases, argv) {
  var __argv = argv || get_argv();
  var __r20 = parse_arguments(__argv, aliases);
  set_argv(__r20.rest);
  delete __r20.rest;
  if (!empty63(__r20)) {
    return __r20;
  }
};
var reload = function(module) {
  delete require.cache[require.resolve(module)];
  return require(module);
};
var run = function(command) {
  return child_process.execSync(command).toString();
};
var __e3 = undefined;
if (typeof(exports) === "undefined") {
  __e3 = {};
} else {
  __e3 = exports;
}
var __exports = __e3;
__exports.readFile = read_file;
__exports.writeFile = write_file;
__exports.fileExists63 = file_exists63;
__exports.directoryExists63 = directory_exists63;
__exports.pathSeparator = path_separator;
__exports.pathJoin = path_join;
__exports.getEnvironmentVariable = get_environment_variable;
__exports.stdout = stdout;
__exports.stderr = stderr;
__exports.write = write;
__exports.exit = exit;
__exports.getArgv = get_argv;
__exports.setArgv = set_argv;
__exports.arguments = arguments;
__exports.reload = reload;
__exports.run = run;
__exports;
